### PR TITLE
Fixed createGeoFabric() to work for all tenants

### PR DESCRIPTION
**Pull request by StoyanTc**
### Purpose of this PR

1. Extended `createGeoFabric()` method to work for all tenants;
2. Adjested the existing JUnit tests

### CODE REVIEW

File: src/main/java/com/c8db/internal/InternalC8DB.java [hunk 1]
```diff
@@ -72,10 +58,9 @@ public ServerRole deserialize(final Response response) throws VPackException {
         };
     }
 
-    protected Request createGeoFabricRequest(final String name, final String spotDc, final String dcList) {
-        final Request request = request(C8RequestParam.DEMO_TENANT, C8RequestParam.SYSTEM, RequestType.POST,
-                InternalC8Database.PATH_API_DATABASE);
-        request.setBody(util().serialize(OptionsBuilder.build(new DBCreateOptions(), name, spotDc, dcList)));
+    protected Request createGeoFabricRequest(final String tenant, final String name, final String spotDc, final String dcList, String getFabric) {
+        final Request request = request(tenant, name, RequestType.POST, InternalC8Database.PATH_API_DATABASE);
```
Comment (tejindersingh-macrometa) [inline src/main/java/com/c8db/internal/InternalC8DB.java:62]: Seems typo. Parameter name should be geoFabric.

**Review by tejindersingh-macrometa — APPROVED** [review]

File: src/main/java/com/c8db/internal/InternalC8DB.java (hunk 1 above)
Comment (StoyanTc) [reply src/main/java/com/c8db/internal/InternalC8DB.java:62]: Thanks. Fixed.

**Review by grainier-macrometa — APPROVED** [review]

File: src/main/java/com/c8db/C8DB.java [hunk 2]
```diff
@@ -695,10 +681,11 @@ public synchronized C8DB build() {
      * @param dcList The list of Edge Locations (Datacenters) as a comma-separated
      *               string. The individual elements for this parameter are your
      *               Edge Location URL prefixes up to the first . character.
+     * @param geoFabric The new of the new db
```
Comment (yehan-macrometa) [inline src/main/java/com/c8db/C8DB.java:684]: Should it be `The name of the geoFabric`?

File: src/main/java/com/c8db/model/DBCreateOptions.java [hunk 3]
```diff
@@ -29,18 +16,25 @@ public DBCreateOptions() {
         super();
     }
 
-    public String getName() {
+    public final String getTenant() {
+		return tenant;
+	}
+
```
Comment (yehan-macrometa) [inline src/main/java/com/c8db/model/DBCreateOptions.java:22]: Let's fix the indentation? Also in line 37 below.

File: src/test/java/com/c8db/internal/velocystream/CommunicationTest.java [hunk 4]
```diff
@@ -129,8 +117,8 @@ public void multiThreadMultiDatabases() throws Exception {
         arangoDB.getVersion();// authentication
 
         try {
-            arangoDB.createGeoFabric(C8Defaults.DEFAULT_TENANT, "db1", "", C8Defaults.DEFAULT_DC_LIST);
-            arangoDB.createGeoFabric(C8Defaults.DEFAULT_TENANT, "db2", "", C8Defaults.DEFAULT_DC_LIST);
+            arangoDB.createGeoFabric(C8Defaults.DEFAULT_TENANT, "db1", "", C8Defaults.DEFAULT_DC_LIST, "db1");
+            arangoDB.createGeoFabric(C8Defaults.DEFAULT_TENANT, "db2", "", C8Defaults.DEFAULT_DC_LIST, "db1");
```
Comment (yehan-macrometa) [inline src/test/java/com/c8db/internal/velocystream/CommunicationTest.java:121]: Should the last param be `db2`?

File: src/main/java/com/c8db/C8DB.java (hunk 2 above)
Comment (StoyanTc) [reply src/main/java/com/c8db/C8DB.java]: Fixed.

File: src/main/java/com/c8db/model/DBCreateOptions.java (hunk 3 above)
Comment (StoyanTc) [reply src/main/java/com/c8db/model/DBCreateOptions.java:21]: Fixed.

File: src/test/java/com/c8db/internal/velocystream/CommunicationTest.java (hunk 4 above)
Comment (StoyanTc) [reply src/test/java/com/c8db/internal/velocystream/CommunicationTest.java]: Fixed.

**Review by yehan-macrometa — APPROVED** [review]

**Review by dlazarenko-macrometa — APPROVED** [review]

**Pull request merged**